python/ql/lib/semmle/python/dataflow/new/internal/DataFlowPrivate.qll

```diff
@@ -313,34 +318,6 @@ module LocalFlow {
       // * `foo = x.foo(); await foo.async_method(); foo.close()` and
       // * `async with x.foo() as foo: await foo.async_method()`.
     )
-    or
-    // Async with var definition
-    //  `async with f(42) as x:`
-    //  nodeFrom is `x`, cfg node
-    //  nodeTo is `x`, essa var
-    //
-    // This makes the cfg node the local source of the awaited value.
-    //
-    // We have this step in addition to the step above, to handle cases where the QL
-    // modeling of `f(42)` requires a `.getAwaited()` step (in API graphs) when not
-    // using `async with`, so you can do both:
-    // * `foo = await x.foo(); await foo.async_method(); foo.close()` and
-    // * `async with x.foo() as foo: await foo.async_method()`.
-    exists(With with, ControlFlowNode var |
-      nodeFrom.(CfgNode).getNode() = var and
-      nodeTo.(EssaNode).getVar().getDefinition().(WithDefinition).getDefiningNode() = var and
-      with.getOptionalVars() = var.getNode() and
-      with.isAsync()
-    )
```

Comment:
I was a little suspicious about this change, but looking over the changes I did in https://github.com/github/codeql/pull/13676/commits/1f93e5b58d8d29a1c9b877cdd8091c23ef9e2419, I honestly don't quite see how this bit was required in the first place :thinking: 

However, since `python/ql/test/library-tests/frameworks/aiohttp/client_request.py` is still passing, I think we're good :+1: